plugins/woocommerce/src/Blocks/Utils/ProductGalleryUtils.php

```diff
@@ -55,13 +95,8 @@ public static function get_product_gallery_images( $post_id, $size = 'full', $at
 					$product_image_html_processor = new \WP_HTML_Tag_Processor( $product_image_html );
 					$product_image_html_processor->next_tag( 'img' );
 					$product_image_html_processor->set_attribute(
```

Comment:
Creating context for the sake of storing an image ID felt unnecessary. I've removed this and instead store this as a data attribute on the element. We need this when we trigger an event on the image, so we have direct access to this via `event.target.getAttribute`. It also removes the confusion of contexts being merged at various levels.